rpc/js/src/dial.ts

```diff
@@ -425,3 +450,42 @@ function iceCandidateToProto(i: RTCIceCandidateInit): ICECandidate {
 	}
 	return candidate;
 }
+
+function validateDialOptions(opts?: DialOptions) {
+	if (!opts) {
+		return;
+	}
+
+	if (opts.accessToken && opts.accessToken.length > 0) {
+		if (opts.authEntity) {
+			throw new Error("cannot set authEntity with accessToken");
+		}
+
+		if (opts.credentials) {
+			throw new Error("cannot set credentials with accessToken");
+		}
+
+		if (opts.externalAuthAddress) {
+			throw new Error("cannot set externalAuthAddress with accessToken");
+		}
+
+		if (opts.externalAuthToEntity) {
+			throw new Error("cannot set externalAuthToEntity with accessToken");
+		}
+
+		if (opts.webrtcOptions) {
```

Comment:
so i think the one change to make here is that you can set a signalingAccessToken as well

Reply:
Im not sure i get the case for separate access token, at least for how things stand today.

If your using signaling/webrtc the access token is only relevant for the signaling server. The actual peer connection doesn't need header anyways. 

If you're direct connecting you're never dealing with a signaling server.

Reply:
it's for the external auth concept which is a part of the rpc utils as it pertains to this being its own open source thing that has more options than we expose in rdk.

the case is your signaling server (external) uses its own auth that is separate from your peer such that you'd send two separate sets of credentials and never comingle the two

Reply:
Updated PR to include that.